tests/test_component_events.py

```diff
@@ -0,0 +1,77 @@
+import collagraph as cg
+
+
+def test_components_events():
+    counter = 0
+    parent = None
+    child = None
+
+    class Parent(cg.Component):
+        def __init__(self, props):
+            super().__init__(props)
+            nonlocal parent
+            parent = self
+
+        def bump(self):
+            nonlocal counter
+            counter += 1
+            self.state["bumped"] = True
+
+        def bump_with_arg(self, step):
+            nonlocal counter
+            counter += step
+            self.state["bumped"] = False
+
+        def render(self):
+            nonlocal counter
+            props = {
+                "on_bump": self.bump,
+                "value": "foo",
+            }
+            if self.state.get("bumped", False):
+                props["on_bump_step"] = self.bump_with_arg
+            return cg.h("parent", {}, cg.h(Child, props))
+
+    class Child(cg.Component):
+        def __init__(self, props):
+            super().__init__(props)
+            nonlocal child
+            child = self
+
+        def simple_event(self):
+            self.emit("bump")
+
+        def event_with_arg(self):
+            self.emit("bump_step", 4)
+
+        def render(self):
+            return cg.h(
+                "child",
+                {
+                    "on_simple_event": self.simple_event,
+                    "on_event_with_arg": self.event_with_arg,
+                    "value": self.props["value"],
+                },
+            )
+
+    gui = cg.Collagraph(event_loop_type=cg.EventLoopType.SYNC)
```

Comment:
I just realized that the default renderer is the dict renderer. Is that really desirable?

Reply:
The dict renderer doesn't pull in any dependencies, so in that sense it is a 'nice' default. But I guess the question is: do we even want a default renderer? Might be better to make it required. If you agree, I'll open another issue for that.

Reply:
I would say it should be a required argument, yeah

Reply:
I've created #41 to track this.